src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/GetLatestVersionUtility.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    public static class GetLatestVersionUtility
+    {
+        /// <summary>
+        /// Return the latest version available in the sources
+        /// </summary>
+        /// <param name="sources">Sources to look at</param>
+        /// <param name="logger">Logger</param>
+        /// <param name="packageId">Package to look for</param>
+        /// <param name="prerelease">Whether to include prerelease versions</param>
+        /// <returns></returns>
+        public static async Task<NuGetVersion> GetLatestVersionFromSources(IList<PackageSource> sources, ILogger logger, string packageId, bool prerelease)
+        {
+            var maxTasks = Environment.ProcessorCount;
+            var tasks = new List<Task<NuGetVersion>>();
+            var latestReleaseList = new List<NuGetVersion>();
+
+            foreach (var source in sources)
+            {
+                tasks.Add(Task.Run(() => GetLatestVersionFromSource(source, logger, packageId, prerelease)));
+                if (maxTasks <= tasks.Count)
+                {
+                    var finishedTask = await Task.WhenAny(tasks);
+                    tasks.Remove(finishedTask);
+                    latestReleaseList.Add(await finishedTask);
+                }
+            }
+
+            await Task.WhenAll(tasks);
+
+            foreach (var t in tasks)
+            {
+                var result = await t;
+                if (result != null)
+                {
+                    latestReleaseList.Add(await t);
+                }
+            }
+
+            return latestReleaseList.Max();
+        }
+
+        /// <summary>
+        /// Return the latest version of the source
+        /// </summary>
+        /// <param name="source">Source to look at</param>
+        /// <param name="logger">Logger</param>
+        /// <param name="packageId">Package to look for</param>
+        /// <param name="prerelease">Whether to include prerelease versions</param>
+        /// <returns></returns>
+        public static async Task<NuGetVersion> GetLatestVersionFromSource(PackageSource source, ILogger logger, string packageId, bool prerelease)
```

Comment:
Since it is async, can you append `Async` to the method name?

```suggestion
        public static async Task<NuGetVersion> GetLatestVersionFromSourceAsync(PackageSource source, ILogger logger, string packageId, bool prerelease)
```